models/stock_picking.py

```diff
@@ -322,14 +324,100 @@ def update_picking(
             # mark_as_done the stock.move.lines
             mls_done = move_lines.mark_as_done(**values)
 
+            if real_time_update:
+                picking = self._real_time_update(mls_done)
+                validate = True
+
         if force_validate or validate:
-            if self.move_line_ids.get_lines_todo() and not create_backorder:
+            if picking.move_line_ids.get_lines_todo() and not create_backorder:
                 raise ValidationError(
                         _('Cannot validate transfer because there'
                           ' are move lines todo'))
             # by default action_done will backorder the stock.move.lines todo
             # validate stock.picking
-            self.action_done() # old do_transfer
+            picking.action_done() # old do_transfer
+
+    def _requires_backorder(self, mls):
+        # if we
+        #   have all the moves
+        #   have all the move_lines within them
+        #   are not waiting for any previous moves to complete
+        # then we dont need to backorder
```

Comment:
The above comments are just pseudocode and don't make the code more readable. I'd remove that and consider adding a docstring explaining what the method does (instead of how).